src/fiatconnect/saga.ts

```diff
@@ -445,14 +459,54 @@ export function* _getSpecificQuote({
     providerIds: [providerId],
   })
   const normalizedQuotes = normalizeFiatConnectQuotes(flow, quotes)
-  const normalizedQuote = normalizedQuotes.find((q) => q.getFiatAccountType() === fiatAccountType)
+
+  // If no account was provided, we need to check cached accounts for a possible match, and fetch the obfuscated account
+  // from the selected provider.
+  let selectedFiatAccount = fiatAccount
+  if (!selectedFiatAccount) {
```

Comment:
nit: could be a bit cleaner to deal with this case higher up, where this function gets called. Looks like `handleAttemptReturnUserFlow` already has its own logic for getting the fiat account, so this logic should only apply to `handleRefetchQuote`. It could also be put into its own helper function for readability and testability